c2cgeoportal/lib/__init__.py

```diff
@@ -224,25 +222,19 @@ def get_typed(name, value, types, request, errors):
 def add_url_params(url, params):
     if len(params.items()) == 0:
         return url
-    return add_spliturl_params(urlsplit(url), params)
+    return add_spliturl_params(urlparse.urlsplit(url), params)
 
 
 def add_spliturl_params(spliturl, params):
-    query = []
-    if spliturl.query != "":
-        query.append(spliturl.query)
-    prepared_params = []
-    for param in params.items():
-        if isinstance(param[1], unicode):
-            prepared_params.append([param[0], param[1].encode("utf-8")])
-        else:
-            prepared_params.append(param)
-
-    query.extend([urllib.urlencode(dict([param])) for param in prepared_params])
+    query = dict([(k, v[-1]) for k, v in urlparse.parse_qs(
```

Comment:
You use the same mechanisme 3 time. Perhaps that's more lisible with a lambda function:
`encode = lambda val: val.encode('utf-8') if isinstance(val, unicode) else val`